pkg/security/podsecuritypolicy/provider.go

```diff
@@ -20,24 +20,24 @@ import (
 	"fmt"
 	"strings"
 
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
```

Comment:
I'm may be wrong, but I think this is required. If I'm mistaken, please correct me.

For example, one of problems I had is [this function](https://github.com/kubernetes/kubernetes/pull/67846/files#diff-291b8dd7d08cc034975ddb3925dbb08fR341). As lister is changed to external one, it now must take `policyv1beta1` external type instead of internal type.

However, that function invokes [`psp.NewSimpleProvider`](https://github.com/kubernetes/kubernetes/pull/67846/files#diff-291b8dd7d08cc034975ddb3925dbb08fR350), which takes internal policy type. Invoking this function with external policy causes type mismatch.

If I update `psp.NewSimpleProvider` to use external type, I must update all other related functions to use external type, or otherwise I'll run into a big number of issues and errors. And therefore, most of PSP-related things are moved to external type in this PR.

My guess is that this is okay. I'm unsure is there a better way.

Reply:
@sttts ^^^ Is it ok?